--- a/src/mongo/util/processinfo_linux.cpp
+++ b/src/mongo/util/processinfo_linux.cpp
@@ -46,9 +46,9 @@
 #include <sys/time.h>
 #include <sys/utsname.h>
 #include <unistd.h>
-#ifdef __UCLIBC__
+
 #include <features.h>
-#else
+#ifdef __GLIBC__
 #include <gnu/libc-version.h>
 #endif
 
@@ -656,11 +656,11 @@
 
     BSONObjBuilder bExtra;
     bExtra.append("versionString", LinuxSysHelper::readLineFromFile("/proc/version"));
-#ifdef __UCLIBC__
+#if defined(__UCLIBC__)
     stringstream ss;
     ss << "uClibc-" << __UCLIBC_MAJOR__ << "." << __UCLIBC_MINOR__ << "." << __UCLIBC_SUBLEVEL__;
     bExtra.append("libcVersion", ss.str());
-#else
+#elif defined(gnu_get_libc_version)
     bExtra.append("libcVersion", gnu_get_libc_version());
 #endif
     if (!verSig.empty())
